### ISSUE TITLE
Tool use for fireworks.ai seem to be broken

### ISSUE THREAD

**Issue by CVxTz** [langchain-ai/langchain#27847]
### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the LangChain documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I am sure that this is a bug in LangChain rather than my code.
- [X] The bug is not resolved by updating to the latest stable version of LangChain (or the specific integration package).

### Example Code

```
import os
from pathlib import Path

from dotenv import load_dotenv
from langchain_community.tools import WikipediaQueryRun
from langchain_community.utilities import WikipediaAPIWrapper
from langchain_core.messages import HumanMessage
from langchain_fireworks import ChatFireworks

load_dotenv(dotenv_path=Path(__file__).parents[1] / ".env")

LLM_API_KEY = os.environ.get("LLM_API_KEY") # Put api key here if no .env exists


wikipedia = WikipediaQueryRun(api_wrapper=WikipediaAPIWrapper())

tools = [wikipedia]


client_medium = ChatFireworks(
    api_key=LLM_API_KEY,
    model="accounts/fireworks/models/llama-v3p1-8b-instruct",
    temperature=0,
)

llm_with_tools = client_medium.bind_tools(tools, tool_choice="wikipedia") # or tool_choice="any"
result = llm_with_tools.invoke([HumanMessage(content="What is stable diffusion")])

print(result.tool_calls) # returns [] 
```

### Description

I am working with the fireworks.ai client and I noticed that Llama 8B never uses a tool even when tool_choice is equal to the tool name or to "any" to force a tool use. This contradicts the documentation for this parameter (https://python.langchain.com/docs/how_to/tool_choice/ ) but I am not sure why it is happening.

Thanks ! 

### System Info

System Information
------------------
> OS:  Linux
> OS Version:  #47~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Wed Oct  2 16:16:55 UTC 2
> Python Version:  3.11.10 (main, Oct  3 2024, 07:29:13) [GCC 11.2.0]

Package Information
-------------------
> langchain_core: 0.3.15
> langchain: 0.3.6
> langchain_community: 0.3.4
> langsmith: 0.1.139
> langchain_fireworks: 0.2.5
> langchain_openai: 0.2.5
> langchain_text_splitters: 0.3.1
> langgraph: 0.2.39

Optional packages not installed
-------------------------------
> langserve

Other Dependencies
------------------
> aiohttp: 3.10.10
> async-timeout: Installed. No version info available.
> dataclasses-json: 0.6.7
> fireworks-ai: 0.15.7
> httpx: 0.27.2
> httpx-sse: 0.4.0
> jsonpatch: 1.33
> langgraph-checkpoint: 2.0.2
> langgraph-sdk: 0.1.35
> numpy: 1.26.4
> openai: 1.53.0
> orjson: 3.10.10
> packaging: 24.1
> pydantic: 2.8.2
> pydantic-settings: 2.6.1
> PyYAML: 6.0.2
> requests: 2.32.3
> requests-toolbelt: 1.0.0
> SQLAlchemy: 2.0.36
> tenacity: 9.0.0
> tiktoken: 0.8.0
> typing-extensions: 4.12.2

**Comment by aldrinjenson** [comment]
Yeah. I've found llama from fireworks to be pretty hesistat to use tool calling as well. a good fix would be appreciated!
